src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFinderScoring.cpp

```diff
@@ -652,23 +649,24 @@ namespace OpenMS
         transition_group_detection.getLibraryIntensity(normalized_library_intensity);
         OpenSwath::Scoring::normalize_sum(&normalized_library_intensity[0], boost::numeric_cast<int>(normalized_library_intensity.size()));
         std::vector<std::string> native_ids_detection;
-        std::string precursor_id;
+        std::vector<std::string> precursor_ids;
         for (Size i = 0; i < transition_group_detection.size(); i++)
         {
           native_ids_detection.push_back(transition_group_detection.getTransitions()[i].getNativeID());
         }
         for (Size i = 0; i < transition_group_detection.getPrecursorChromatograms().size(); i++)
         {
-          // try to identify the correct precursor native id
           String precursor_chrom_id = transition_group_detection.getPrecursorChromatograms()[i].getNativeID();
-          if (OpenSwathHelper::computePrecursorId(transition_group.getTransitionGroupID(), 0) == precursor_chrom_id)
+
+          // append precursor monoisotopic peak and additional isotopes with a signal for scoring
+          if (OpenSwathHelper::computePrecursorId(transition_group.getTransitionGroupID(), 0) == precursor_chrom_id || mrmfeature->getPrecursorFeature(precursor_chrom_id).getIntensity() > 0)
```

Comment:
We always score the precursor monoisotopic peak and only those isotopes that have a signal.

Reply:
hmm, I wonder whether that is really fair because you then compare precursors with 3 isotopes to those with only 1 -- also it will lead to a situation where when you have a tiny signal your score becomes much worse...

Reply:
I will have a closer look at this. The problem is that on MS1-level the signal is much more sparse (or clean) than on MS2-level in the Orbitrap data.

Reply:
yes, we probably have to look into this more -- the MS1 signal being "cleaner" could be an artefact of low fill time?